cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -201,8 +212,8 @@ private Collection<AlterConfigOp> diff(int brokerId, String desired,
         return updatedCE;
     }
 
-    private void updateOrAdd(String propertyName, Map<String, ConfigModel> configModel, Map<String, String> desiredMap, Collection<AlterConfigOp> updatedCE) {
-        if (!isIgnorableProperty(propertyName)) {
+    private void updateOrAdd(String propertyName, Map<String, ConfigModel> configModel, Map<String, String> desiredMap, Collection<AlterConfigOp> updatedCE, boolean isCombined) {
```

Comment:
It's because the KafkaBrokerConfigurationDiff class is only initialized if we are considering the configuration of a broker. So if `nodeRef.controller()` is also true, then it means this is a combined node.

Reply:
I think that creates a bad code that someone badly reuses down the road.

Reply:
We could update the call to that function with `nodeRef.broker() && nodeRef.controller()`? Or just pass the nodeRef and do the check inside the isIgnorableProperty function

Reply:
Tina updated it to explicitly check both `nodeRef.broker()` and `nodeRef.controller()` in the calling functions